rl/core.py

```diff
@@ -54,23 +54,23 @@ def fit(self, env, nb_steps, action_repetition=1, callbacks=None, verbose=1,
 
         episode = 0
         self.step = 0
-        observation = None
+        self.observation = None
```

Comment:
I don't really like making this stateful. Can't the SARSA agent keep track of this internally? I'd use something like this:

```
def reset_states(self):
    # the other stuff as well + the 3 lines
    self.actions = deque(maxlen=2)
    self.observations = deque(maxlen=2)
    self.rewards = deque(maxlen=2)

def forward(self, observation):
    self.observations.append(observation)
    # TODO: do the other stuff to get action
    self.actions.append(action)

def backward(self, reward, terminal):
    self.rewards.append(reward)
    if len(self.observations) < 2:
        return  # not enough data yet
    
    # Otherwise, get SARSA stuff for SARSA:
    state0 = self.observations[0]
    action0 = self.actions[0]
    reward = self.rewards[0]
    state1 = self.observations[1]
    action1 = self.actions[1]

    # TODO: and use it to update the network like before.
```

What do you think?